src/nspawn/nspawn.c

```diff
@@ -3841,6 +3857,20 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
+        if (arg_syscall_whitelist != NULL) {
+                const char *p = arg_syscall_whitelist;
+                for (;;) {
+                        _cleanup_free_ char *t = NULL;
+                        r = extract_first_word(&p, &t, ",", 0);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse syscall %s.", t);
+                        if (r == 0)
+                                break;
+                        strcat(t, "\0");
```

Comment:
We should get rid of `strcat`, but that will be most likely done by using `strv_split_extract`